pkg/handlers/embedded_cluster_artifacts.go

```diff
@@ -182,3 +239,71 @@ func (h *Handler) GetEmbeddedClusterK0sImages(w http.ResponseWriter, r *http.Req
 		return
 	}
 }
+
+// createECChartsArchive creates a .tgz archive of the given embedded cluster charts directory
+func createECChartsArchive(chartsDir string) (string, error) {
+	// Create temporary .tgz file
+	tmpFile, err := os.CreateTemp("", "ec-charts-*.tgz")
+	if err != nil {
+		return "", errors.Wrap(err, "failed to create temporary file")
+	}
+	defer tmpFile.Close()
+
+	// Create gzip writer
+	gzipWriter := pgzip.NewWriter(tmpFile)
+	defer gzipWriter.Close()
+
+	// Create tar writer
+	tarWriter := tar.NewWriter(gzipWriter)
+	defer tarWriter.Close()
+
+	err = filepath.Walk(chartsDir, func(filePath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Skip directories
+		if info.IsDir() {
+			return nil
+		}
+
+		// Get relative path for tar header name
+		relPath, err := filepath.Rel(chartsDir, filePath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get relative path for %s", filePath)
+		}
+
+		// Create tar header
+		header := &tar.Header{
+			Name:    relPath,
+			Mode:    int64(info.Mode()),
+			Size:    info.Size(),
+			ModTime: info.ModTime(),
+		}
+
+		// Write header
+		if err := tarWriter.WriteHeader(header); err != nil {
+			return errors.Wrap(err, "failed to write tar header")
+		}
+
+		// Open and copy file contents
+		chartFile, err := os.Open(filePath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to open file %s", filePath)
+		}
+		defer chartFile.Close()
+
+		// Copy file data into tar writer
+		if _, err := io.Copy(tarWriter, chartFile); err != nil {
+			return errors.Wrapf(err, "failed to copy file %s to tar archive", filePath)
+		}
+
+		return nil
+	})
+
+	if err != nil {
```

Comment:
you should remove the file here

Reply:
why? the file gets removed at the end of the handler after streaming.

Reply:
oh nevermind. i see what you're saying.

Reply:
updated. ty.